prime-router/src/main/kotlin/cli/CommandUtilities.kt

```diff
@@ -20,19 +47,31 @@ class CommandUtilities {
         /**
          * The API endpoint to check for.  This needs to be a simple operation.
          */
-        private const val waitForApiEndpointPath = "api/lookuptables/list"
+        const val waitForApiEndpointPath = "api/lookuptables/list"
+
+        /**
+         * timeout for http calls
+         */
+        private const val TIMEOUT = 50_000
+        const val REQUEST_TIMEOUT_MILLIS: Long = 130000 // need to be public to be used by inline
+        const val SETTINGS_REQUEST_TIMEOUT_MILLIS = 30000
 
         /**
          * Waits for the endpoint at [environment] to become available. This function will retry [retries] number of
          * times waiting [pollIntervalSecs] seconds between retries.
          * @throws IOException if a connection was not made
          */
-        internal fun waitForApi(environment: Environment, retries: Int = 30, pollIntervalSecs: Long = 1) {
+        internal fun waitForApi(
+            environment: Environment,
+            retries: Int = 30,
+            pollIntervalSecs: Long = 1,
+            httpClient: HttpClient? = null,
```

Comment:
> Curious as to why we would want to allow null here? That seems like it would be a developer mistake...

a named arg with default null will make callers simple - mostly just with the required url = <url-val>

the httpClient is added as a named arg - for now - it is only used for mock tests where an ktor mock engine instrumented client is injected, so that the logic can be tested.

Reply:
I guess I'm wondering why we wouldn't default it to a real client rather than making all downstream code handle the null case?